posthog/models/person/sql.py

```diff
@@ -520,3 +520,23 @@
 
 GET_PERSON_COUNT_FOR_TEAM = "SELECT count() AS count FROM person WHERE team_id = %(team_id)s"
 GET_PERSON_DISTINCT_ID2_COUNT_FOR_TEAM = "SELECT count() AS count FROM person_distinct_id2 WHERE team_id = %(team_id)s"
+
+
+CREATE_PERSON_DISTINCT_ID_OVERRIDES_DICTIONARY = """
+CREATE OR REPLACE DICTIONARY {database}.person_distinct_id_overrides_dict ON CLUSTER {cluster} (
+    `team_id` Int64, -- team_id could be made hierarchical to save some space.
+    `distinct_id` String,
+    `person_id` UUID
+)
+PRIMARY KEY team_id, distinct_id
+-- For our own sanity, we explicitly write out the group by query.
+SOURCE(CLICKHOUSE(
+    query 'SELECT team_id, distinct_id, argMax(person_id, version) AS person_id FROM {database}.person_distinct_id_overrides GROUP BY team_id, distinct_id'
+))
+LAYOUT(complex_key_hashed())
+-- ClickHouse will choose a time uniformly within 1 to 5 hours to reload the dictionry (update if necessary to meet SLAs).
+LIFETIME(MIN 3600 MAX 18000)
```

Comment:
If the unit for this setting is in seconds, this means that we're only refreshing the dictionary every 1 to 3 hours? Might be OK for the squash use case, but if we want to make it available to the read path, my instinct is that we want a more frequent refresh?
I'd be happy to start at a low frequency to confirm lack of impact on resources, then tune the frequency up progressively, just want to confirm what the ideal target would be here.

Reply:
The squash forces a refresh, so it pretty much ignores this value (in fact, the squash was creating a dictionary with `lifetime 0`, i.e. never refreshes)

This is not used by the read path yet so I didn't put much thought into what the read path lifetime would be. That being said, if/when the read path starts querying this dict, then yeah, this refresh time will need to account any desired SLA. I am not sure what this SLA would be though (or if one even exists). Do you have any target frequency in mind?

I'd be happy to bump this frequency up, while running the squash it didn't seem like the dictionary had any performance impact.

Reply:
Shall we bring it down to `MIN 1800 MAX 5400` (30 to 90 minutes)? No particular preference, just imagined that "around an hour" is easy to communicate.

Reply:
If we use this dict in insights, this means that `identify` will only be fully effective after the dict is refreshed. My first instinct would be to target 5 to 10 minutes as an hour seems to slow to me, especially when someone is testing instrumentation changes.

Not sure whether that'd be acceptable through, adding the dict and slowing increasing the frequency while checking CH resources might be the best way to evaluate this?

Reply:
I agree. We can start with 1-3 hours and work our way down to 5-10 minutes 👍